pkg/iam/oidc/api_test.go

```diff
@@ -239,6 +240,70 @@ var _ = Describe("EKS/IAM API wrapper", func() {
 		})
 
 	})
+
+	Describe("OIDC AWS partition test", func() {
+		var (
+			provider *mockprovider.MockProvider
+			srv      *testServer
+		)
+
+		BeforeEach(func() {
+			provider = mockprovider.NewMockProvider()
+			var err error
+			srv, err = newServer("localhost:10028")
+			Expect(err).NotTo(HaveOccurred())
+			go srv.serve()
+		})
+
+		AfterEach(func() {
+			srv.close()
+		})
+
+		DescribeTable("AssumeRolePolicyDocument", func(partition, expectedAudience string) {
```

Comment:
These table tests are a little bit cryptic. Can you add a better explanation of what this test is checking in its description?

Reply:
I added more description to the tests. LMK what you think.